lib/livebook_web/plugs/configured_plug.ex

```diff
@@ -0,0 +1,17 @@
+defmodule LivebookWeb.ConfiguredPlug do
+  @moduledoc false
+
+  # Runs plugs configured for the :livebook application
+
+  @behaviour Plug
+
+  @impl true
+  def init(_opts) do
+    plugs = Application.get_env(:livebook, :plugs, [])
```

Comment:
This will be resolved at compilation time. We probably want no-op on init and then read it on `call`.

Reply:
Also keep in mind the plugs in the router only run if there is a matching route. So if we want people to add "custom pages", then we need to move the plug call to the endpoint.

Reply:
Yeah, that's one more reason to move the plug to the top. The users may want to access session, but on the other hand placing the plug in the middle is kinda arbitrary too. And I think it's more likely that the use case is to setup some initial-level handling like nerves, rather than overriding the web app. So maybe let's make it the first plug in the endpoint, If we ever need both, we can make this configurable too, wdyt?

Reply:
I think doing it right before the router is not arbitrary and a bit more useful than at the top. :)

Reply:
I meant calling in the middle of the router is arbitrary ^^ Before router it is then!